v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/SpannerToSourceDbITBase.java

```diff
@@ -0,0 +1,219 @@
+import org.apache.beam.it.common.utils.PipelineUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.artifacts.utils.ArtifactUtils;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.beam.it.gcp.spanner.SpannerResourceManager;
+import org.apache.beam.it.gcp.storage.GcsResourceManager;
+import org.apache.beam.it.jdbc.JDBCResourceManager;
+import org.apache.beam.it.jdbc.MySQLResourceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SpannerToSourceDbITBase extends TemplateTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SpannerToSourceDbITBase.class);
+
+  protected SpannerResourceManager createSpannerDatabase(String spannerSchemaFile)
+      throws IOException {
+    SpannerResourceManager spannerResourceManager =
+        SpannerResourceManager.builder("rr-main-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build();
+    String ddl =
+        String.join(
+            " ",
+            Resources.readLines(Resources.getResource(spannerSchemaFile), StandardCharsets.UTF_8));
+    ddl = ddl.trim();
+    String[] ddls = ddl.split(";");
+    for (String d : ddls) {
+      if (!d.isBlank()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    return spannerResourceManager;
+  }
+
+  protected SpannerResourceManager createSpannerMetadataDatabase() throws IOException {
+    SpannerResourceManager spannerMetadataResourceManager =
+        SpannerResourceManager.builder("rr-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build();
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
+    return spannerMetadataResourceManager;
+  }
+
+  public PubsubResourceManager setUpPubSubResourceManager() throws IOException {
+    return PubsubResourceManager.builder(testName, PROJECT, credentialsProvider).build();
+  }
+
+  public SubscriptionName createPubsubResources(
+      String identifierSuffix, PubsubResourceManager pubsubResourceManager, String gcsPrefix) {
+    String topicNameSuffix = "rr-it" + identifierSuffix;
+    String subscriptionNameSuffix = "rr-it-sub" + identifierSuffix;
+    TopicName topic = pubsubResourceManager.createTopic(topicNameSuffix);
+    SubscriptionName subscription =
+        pubsubResourceManager.createSubscription(topic, subscriptionNameSuffix);
+    String prefix = gcsPrefix;
+    if (prefix.startsWith("/")) {
+      prefix = prefix.substring(1);
+    }
+    prefix += "/retry";
+    gcsClient.createNotification(topic.toString(), prefix);
+    return subscription;
+  }
+
+  protected void createAndUploadShardConfigToGcs(
+      GcsResourceManager gcsResourceManager, MySQLResourceManager jdbcResourceManager)
+      throws IOException {
+    Shard shard = new Shard();
+    shard.setLogicalShardId("Shard1");
```

Comment:
It might be better to send shard id from the actual test and not hard code it here

Reply:
Thanks. Except multishard, all have the same shard hence kept like so, but yea it can also be parameterized. For now, keep as is though.